.githooks/pre-commit

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```

Comment:
Considering we do not use pre-commit hooks in any other places, would it be better to replace this with an update/verify script that can be called through a Makefile target and verified as part of `make verify`?

Reply:
Yep, no problem.
But the original discussion about this is to incorporate it into CI. I can not figure it out how is the CI run in this project. That's why I made it as a git hook. Making it into hack script makes little sense.

The proper logic should be, any update submitted will be checked by the CI with some scripts, then the submitter will be alerted to update it with verify script if it fails.

`make verify` makes some sense. It can be used properly and flexibly. 
1. make verify
2. update.sh/make update docs
I will make it proper.

Reply:
I have updated it with `make verify`.

Reply:
Rebase work has been done.